base/sort.jl

```diff
@@ -1160,8 +1160,8 @@ function sortperm(A::AbstractArray;
         n = length(A)
         if n > 1
             min, max = extrema(A)
-            (diff, o1) = sub_with_overflow(max, min)
-            (rangelen, o2) = add_with_overflow(diff, oneunit(diff))
+            (diff, o1) = sub_with_overflow(max, min)::Tuple{Any, Bool}
+            (rangelen, o2) = add_with_overflow(diff, oneunit(diff))::Tuple{Any, Bool}
             if !o1 && !o2 && rangelen < div(n,2)
```

Comment:
Is this equally as effective?
```suggestion
            (diff, o1) = sub_with_overflow(max, min)
            (rangelen, o2) = add_with_overflow(diff, oneunit(diff))
            if !(o1 || o2) && rangelen < div(n,2)
```

Reply:
I just checked it - sadly, this does not help :disappointed:

Reply:
Oh, right. If `o1` is `false` and `o2` is `Static.False` then this would still dispatch to `!(::False)`. What about
```suggestion
            (diff, o1) = sub_with_overflow(max, min)
            (rangelen, o2) = add_with_overflow(diff, oneunit(diff))
            if !(o1 || o2)::Bool && rangelen < div(n,2)
```
That should never call `!(::False)`.